src/game/Game.kt

```diff
@@ -2,31 +2,35 @@ package game
 
 import action.Action
 import agent.Player
+import exception.ExceptionMessages.Companion.HIT_WALL_EXCEPTION_MESSAGE
+import exception.ExceptionMessages.Companion.WALL_FEEDBACK
+import exception.InvalidMoveException
 import map.ExitNode
 import map.WorldMap
-import java.lang.UnsupportedOperationException
 
 class Game(private val player: Player, private val worldMap: WorldMap) {
 
     var timeCounter: Int = 0
 
     fun run() {
         while(!Thread.interrupted()) {
-            val nextMove = player.chooseNextMove()
-            when (nextMove) {
-                Action.MOVE_NORTH ->
-                    worldMap.moveObject(player.actor, 0, -1)
-                else -> {
-                    throw UnsupportedOperationException()
-                }
+            when (player.chooseNextMove()) {
+                Action.MOVE_NORTH -> moveNorthLogic()
             }
-
             if (!player.actor.alive || worldMap.getNode(worldMap.positionFor(player.actor)) is ExitNode) {
                 break
             }
-
             timeCounter++
         }
     }
 
+    private fun moveNorthLogic() {
+        try {
+            worldMap.moveObject(player.actor, 0, -1)
+        } catch (ex: InvalidMoveException) {
+            when (ex.message) {
```

Comment:
you can just give the exact exception type directly to player.

Reply:
fine